<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="launch_rviz" default="true" />
	<arg name="cloud_topic" default="/ouster_points_self_filtered" />
	<arg name="play_delay" default="0.4" />
	<arg name="play_rate" default="0.5" />
	<arg name="use_sim_time" default="true" />

	<arg name="parameter_file_path" default="$(find m545_volumetric_mapping)/param/params_ouster_os0_128.yaml"/>
	<arg name="parameter_proj_file_path" default="$(find m545_volumetric_mapping)/param/params_projection.yaml"/>
	<arg name="bag_filename" default="filtered_cloud_and_image.bag" />
	<arg name="bag_filepath"
		default="$(find m545_volumetric_mapping)/data/$(arg bag_filename)" />

	<include file="$(find m545_volumetric_mapping)/launch/odometry.launch">
		<arg name="cloud_topic" value="$(arg cloud_topic)"/>
		<arg name="launch_rviz" value="$(arg launch_rviz)"/>
		<arg name="parameter_file_path" value="$(arg parameter_file_path)"/>

	</include>

	<node pkg="rosbag" type="play" name="rosbag"
		args="'$(arg bag_filepath)' --delay '$(arg play_delay)' --rate '$(arg play_rate)' --clock -s 0" />
	<!-- append - -loop for looping -->



</launch>
